lib/sla.js

```diff
@@ -65,16 +91,29 @@ function addChannelsToBridge(channel, bridge) {
  *   addChannelsToBridge().
  * @return {Q} Q promise object.
  */
-function createChannel(client, bridge) {
+function createChannel(client, inbound, bridge) {
   var defer = Q.defer();
-  var channel = client.Channel();
-  var originate = Q.denodeify(channel.originate.bind(channel));
-  originate({endpoint: 'SIP/phone', app: 'sla'})
+  var dialed = client.Channel();
+  var originate = Q.denodeify(dialed.originate.bind(dialed));
+  originate({endpoint: 'SIP/phone', app: 'sla', appArgs: 'dialed'})
     .catch(function (err) {
       defer.reject(err);
     });
-  channel.once('StasisStart', function(event, channel) {
-    defer.resolve(channel);
+  dialed.on('StasisStart', function(event, dialed) {
+    var answer = Q.denodeify(dialed.answer.bind(dialed));
+    answer();
+    defer.resolve(dialed);
+  });
+  dialed.on('ChannelDestroyed', function(event, dialed) {
+    defer.resolve(null);
+  });
+  inbound.on('ChannelHangupRequest', function(event, inbound) {
+    defer.resolve(null);
+    var hangup = Q.denodeify(dialed.hangup.bind(dialed));
+    hangup()
+      .catch(function (err) {
+        defer.reject(err);
```

Comment:
You've already resolved the deferred before this, so this won't ever get called. You should resolve in the success callback of the .done clause so it only runs if catch wasn't called.